apps/angular/permissions/src/app/user.store.ts

```diff
@@ -9,6 +9,25 @@ export class UserStore {
   private user = new BehaviorSubject<User | undefined>(undefined);
   user$ = this.user.asObservable();
 
+  isLoggedIn$ = this.user$.pipe(map(Boolean));
+  isAdmin$ = this.user$.pipe(map((user) => Boolean(user?.isAdmin)));
+
+  hasAnyRole = (role: Role | Role[]) =>
+    this.user$.pipe(
+      map((user) => {
+        if (!user) {
+          return false;
+        }
+
+        if (user.isAdmin) {
+          return true;
+        }
+
+        const roles: Role[] = Array.isArray(role) ? role : [role];
+        return roles.length === 0 || user.roles.some((r) => roles.includes(r));
+      }),
+    );
```

Comment:
you can use signal now 😜

Reply:
I tried to, but found it difficult to refactor directive using signal. May you show solution with singnals?
I don't know to listen/run login inside of directive while signal changes instead of current component store effect